addons/spectrum/functions/fnc_addSoundSequenceServer.sqf

```diff
@@ -0,0 +1,116 @@
+
+if (isNull _unit || _sounds isEqualTo []) exitWith {
+	diag_log format ["CrowsEW:fnc_addSoundSequenceServer.sqf: Can not add sequence with _unit=%1 or _sounds=%2.", _unit, _sounds]; 
+};
+
+// if the unit already plays a sound sequence stop it first
+private _existingHandle = _unit getVariable[QGVAR(radioSoundHandle), scriptNull];
+if (!isNull _existingHandle) then {
+	terminate _existingHandle;
+	// remove signal 
+	[_unit, "sound"] call FUNC(removeBeaconServer);
+
+	// remove from zeus draw list - We wait with publish update until end of script where we publish anyway for the new draws
+	// private _rmIndex = GVAR(radioTrackingAiUnits) findIf { (_x select 0) == _unit};
+	// GVAR(radioTrackingAiUnits) deleteAt _rmIndex;
+};
+
+private _handle = [_unit, _freq, _range, _sounds, _loop] spawn {
+	params ["_unit", "_freq", "_range", "_sounds", "_loop"];
+
+	private _firstSequence = true;	// Is this our first run through the sequence?
+
+	// add signal to spectrum
+	[_unit, _freq, _range, "sound"] call FUNC(addBeaconServer);
+
+	while {_firstSequence || _loop} do {
+		// loop through sounds and play them in sequence
+		{
+			if (!alive _unit) exitWith { _loop = false; };	// if unit is dead, exit
+
+			// if no sound is given but a delay number, sleep for that number in seconds
+			if (_x isEqualType 1337) then {				
+				sleep _x;
+				continue;	// just sleep and continue to next entry in the sequence
+			};
+
+			private _sound = _x;
+
+			// save currently played sound and when it started
+			_unit setVariable[QGVAR(currentRadioSound), _sound, true];
+			_unit setVariable[QGVAR(currentRadioSoundStartTime), serverTime, true];
+
+			// notify all listeners that sound started
+			private _clientIDs = _unit getVariable[QGVAR(currentRadioSoundListeners), []];;
+			{				
+				[QGVAR(newRadioSoundStarted), [_unit], _x] call CBA_fnc_ownerEvent;				
+			} forEach _clientIDs;
+
+			// get sound length
+			private _soundId = playSoundUI [_sound, 0.0];	// zero volume, just to get the soundId
+			private _soundLength = (soundParams _soundId)#2;
+			stopSound _soundId;	// stop sound right away; we only wanted to get the length
+
+			// sleep for length
+			sleep _soundLength;
+
+			// reset sound variables
+			_unit setVariable[QGVAR(currentRadioSound), objNull, true];
+			_unit setVariable[QGVAR(currentRadioSound), "", true];
+			_unit setVariable[QGVAR(currentRadioSoundStartTime), 0, true];
+
+		} forEach _sounds;
+		_firstSequence = false;
+	};
+
+	// remove signal
+	[_unit, "sound"] call FUNC(removeBeaconServer);
+};
+
+
+// save handler to loop or a way to stop it, on var on unit, as we are on server, we only save locally
+_unit setVariable[QGVAR(radioSoundHandle), _handle];
+
+
+// add to array for drawing indication of what AI units has it enabled
+// GVAR(radioTrackingAiUnits) pushBack [_unit, _voicePack];
+// publicVariable QGVAR(radioTrackingAiUnits);
```

Comment:
Commented out on purpose?

Reply:
Yes. Same reason as for lines 51 and 52.